python/lsst/pipe/tasks/dataFrameActions/_baseDataFrameActions.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+__all__ = ("DataFrameAction",)
+
+from lsst.pex.config import Field, ListField
+from typing import Iterable, Any, Mapping
+
+from ..configurableActions import ConfigurableAction
+
+
+class DataFrameAction(ConfigurableAction):
+    _actionCache: Mapping[int, Any]
+
+    cache = Field(doc="Controls if the results this action should be cached, only works on frozen actions",
+                  dtype=bool, default=False)
+    cacheArgs = ListField(doc="If cache is True, this is a list of argument keys that will be used to "
+                          "compute the cache key in addition to the DataFrameId", dtype=str)
+
+    def __init_subclass__(cls, **kwargs) -> None:
+        cls._actionCache = {}
+
+        def call_wrapper(function):
+            def inner_wrapper(self, dataFrame, **kwargs):
+                dfId = id(dataFrame)
+                extra = []
+                for name in (self.cacheArgs or tuple()):
+                    if name not in kwargs:
+                        raise ValueError(f"{name} is not part of call signature and cant be used for "
+                                         "caching")
+                    extra.append(kwargs[name])
+                extra.append(dfId)
+                key = tuple(extra)
+                if self.cache and self._frozen:
+                    # look up to see if the value is in cache already
+                    if result := self._actionCache.get(key):
+                        print("returning cache")
+                        return result
```

Comment:
Should this print statement still be here?

Reply:
nope, good catch